### PR TITLE
v3.0, dev/reference/configuration: add cluster-configuration doc

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? 

This PR mainly defines the configurations of tidb-cluster.

The table of parameters is collated from [values.yaml](https://github.com/pingcap/tidb-operator/blob/master/charts/tidb-cluster/values.yaml). 
The other text is collated from [configuration](https://github.com/pingcap/tidb-operator/blob/master/docs/operation-guide.md#configuration).

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/1448
https://github.com/pingcap/docs-cn/pull/1582
https://github.com/pingcap/docs-cn/pull/1590
https://github.com/pingcap/docs-cn/pull/1645

### Which version does your change affect? 

dev, v3.0

### Checklist 

- [ ] Add a new file to `TOC.md`
- [ ] No Tab spaces anywhere 
- [ ] Leave a blank line both before and after a code block
- [ ] Keep the first level heading consistent with `title` in metadata
- [ ] Use *four* spaces for each level of indentation except that in `TOC.md`

PTAL, thanks!

### CODE REVIEW

**Comment by yikeke** [comment]
The original Chinese doc seems to be modified by https://github.com/pingcap/docs-cn/pull/1582/ and https://github.com/pingcap/docs-cn/pull/1590/. Please update the corresponding parts in this PR after the two PRs are merged. @TomShawn

**Comment by TomShawn** [comment]
> The original Chinese doc seems to be modified by [pingcap/docs-cn#1582](https://github.com/pingcap/docs-cn/pull/1582) and [pingcap/docs-cn#1590](https://github.com/pingcap/docs-cn/pull/1590). Please update the corresponding parts in this PR after the two PRs are merged. @TomShawn

Sure

**Comment by TomShawn** [comment]
The changes from https://github.com/pingcap/docs-cn/pull/1582 and https://github.com/pingcap/docs-cn/pull/1590 are committed above. PTAL, thanks. @yikeke @DanielZhangQD

File: dev/reference/configuration/tidb-in-kubernetes/cluster-configuration.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+| `pd.replicas` | The number of Pods in PD | `3` |
+| `pd.image` | PD image | `pingcap/pd:v3.0.0-rc.1` |
+| `pd.imagePullPolicy` | The pulling policy for PD image | `IfNotPresent` |
+| `pd.logLevel` | The log level of PD<br>If the version of TiDB Operator is later than v1.0.0-beta.3, you can configure the parameter via `pd.config`: <br>`[log]`<br>`level = "info"` | `info` |
```
Comment (DanielZhangQD) [inline dev/reference/configuration/tidb-in-kubernetes/cluster-configuration.md]: "you can configure" to "you must configure" and all the following should be changed.

File: dev/reference/configuration/tidb-in-kubernetes/cluster-configuration.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+| `tidb.preparedPlanCacheEnabled` | Whether to enable TiDB's prepared plan cache<br>If the version of TiDB Operator is later than v1.0.0-beta.3, you can configure this parameter via `tidb.config`:<br>`[prepared-plan-cache]`<br>`enabled = false` | `false` |
+| `tidb.preparedPlanCacheCapacity` | The cache capacity of TiDB's prepared plan<br>If the version of TiDB Operator is later than v1.0.0-beta.3, you can configure this parameter via `tidb.config`:<br>`[prepared-plan-cache]`<br>`capacity = 100`  | `100` |
+| `tidb.txnLocalLatchesEnabled` | Whether to enable the memory lock for transaction. It is recommended to enable the lock when there are many conflicts among local transactions.<br>If the version of TiDB Operator is later than v1.0.0-beta.3, you can configure this parameter via `tidb.config`:<br>`[txn-local-latches]`<br>`enabled = false` | `false` |
+| `tidb.txnLocalLatchesCapacity` |  The capacity of the local latch of a transaction<br>If the version of TiDB Operator is later than v1.0.0-beta.3, you can configure this parameter via `tidb.config`:<br>`[txn-local-latches]`<br>`capacity = 10240000` | `10240000` |
```
Comment (DanielZhangQD) [inline dev/reference/configuration/tidb-in-kubernetes/cluster-configuration.md]: https://github.com/pingcap/docs-cn/pull/1590 reverted some changes in https://github.com/pingcap/docs-cn/pull/1582 and I have fixed them in https://github.com/pingcap/docs-cn/pull/1643, please help sync the change.

File: dev/reference/configuration/tidb-in-kubernetes/cluster-configuration.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+| `pd.config` | The configuration of PD. Check [this link](https://github.com/pingcap/pd/blob/master/conf/config.toml) for the file of the default PD configuration (by choosing the tag of the corresponding PD version). You can see [this document](https://pingcap.com/docs-cn/v3.0/reference/configuration/pd-server/configuration-file/) for the detailed description of the configuration parameters (by choosing the corresponding document version). Here you only need to **modify the configuration based on the format of the configuration file**.  | If the version of TiDB Operator is v1.0.0-beta.3 or earlier, the default value is <br>`nil`<br>If the version of TiDB Operator is later than v1.0.0-beta.3, the default value is <br>`[log]`<br>`level = "info"`<br>`[replication]`<br>`location-labels = ["region", "zone", "rack", "host"]`.<br>Example of configuration:<br>&nbsp;&nbsp;`config:` \|<br>&nbsp;&nbsp;&nbsp;&nbsp;`[log]`<br>&nbsp;&nbsp;&nbsp;&nbsp;`level = "info"`<br>&nbsp;&nbsp;&nbsp;&nbsp;`[replication]`<br>&nbsp;&nbsp;&nbsp;&nbsp;`location-labels = ["region", "zone", "rack", "host"]` |
```
Comment (DanielZhangQD) [inline dev/reference/configuration/tidb-in-kubernetes/cluster-configuration.md]: Maybe it's better to replace "you only need to" with "you have to", users should follow the format in configuration file exactly.

File: dev/reference/configuration/tidb-in-kubernetes/cluster-configuration.md (hunk 2 above)
Comment (TomShawn) [reply dev/reference/configuration/tidb-in-kubernetes/cluster-configuration.md]: Sure

**Pull request merged**